server/src/Irony.cpp

```diff
@@ -428,12 +471,28 @@ void Irony::candidates() const {
       // https://github.com/Sarcasm/irony-mode/pull/78#issuecomment-37115538
       if (chunkKind == CXCompletionChunk_TypedText && !typedTextSet) {
         typedtext = chunk.text();
+        if (!startsWith(typedtext, prefix, caseInsensitive)) {
+          hasPrefix = false;
+          break;
+        }
         // annotation is what comes after the typedtext
         annotationStart = prototype.size();
         typedTextSet = true;
       }
     }
 
+    if (!hasPrefix) {
+      continue;
+    }
+    if (!typedTextSet) {
```

Comment:
I can reproduce it in clang3.9.1:
```c++
void foobar(int parameter)
{
}

int main(int argc, char *argv[])
{
    foobar(xxx <-complete here
}

```
Without the check, output of *Irony* (I disabled flycheck-mode) is:

```elisp
(
  ("" 1 "void" "" "foobar(int parameter)" 0 ("") available)
)

;;EOT

```